### PR TITLE
Add db and pool Metrics Swagger coverage and examples

### PR DESCRIPTION
**Pull request by nathannfan**
Add database and elastic pool metric swagger coverage and examples.  Move usages to usages.json.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@nathannfan Please add security definitions as described [here](https://github.com/Azure/autorest/blob/master/docs/developer/validation-rules/security-definitions-structure-validation.md) (see https://github.com/Azure/azure-rest-api-specs/search?q=securityDefinitions for typical structure within Swagger) to *all* of your Swaggers 🙂

**Comment by olydis** [comment]
not sure about your SDK/CLI release state, but be aware of the breaking changes (at least due to renaming of types) introduced here

**Comment by nathannfan** [comment]
@olydis Jared added them to all of them already, so I added them to my new file.

**Comment by nathannfan** [comment]
@olydis not sure why I'm failing validation for ruby.  Any ideas?

**Comment by jaredmoo** [comment]
I think the ruby build is broken

**Comment by olydis** [comment]
@jaredmoo seems so :( will check with the Ruby people, but we can go on here independently for the time being

File: arm-sql/2014-04-01/swagger/metrics.json [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The name of the database."
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "An OData filter expression that describes a subset of metrics to return.  Required for getting metrics."
```
Comment (olydis) [inline arm-sql/2014-04-01/swagger/metrics.json]: General guideline: don't repeat things like requiredness in the description. For generated SDKs, imagine a method `ListMetrics` with a regular (non-optional) parameter `filter` with a description `Required for getting metrics.`. You would probably think "Yeah, obviously..."

File: arm-sql/2014-04-01/swagger/metrics.json [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The name of the elastic pool."
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "An OData filter expression that describes a subset of metrics to return.  Required for getting metrics."
```
Comment (olydis) [inline arm-sql/2014-04-01/swagger/metrics.json]: see above, remove `Required for getting metrics.`

File: arm-sql/2014-04-01/swagger/metrics.json [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+          "type": "string",
+          "description": "The unit of the metric.",
+          "enum": ["count", "bytes", "seconds", "percent", "countPerSecond", "bytesPerSecond"],
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "UnitType"
+          }
+        },
+        "name": {
+          "$ref": "#/definitions/MetricName",
```
Comment (olydis) [inline arm-sql/2014-04-01/swagger/metrics.json]: think you wanna tag those `$ref`ed properties as `readOnly` too :)

**Review by olydis — APPROVED** [review]
minor comments. as always, very high quality! 👍

**Review by jaredmoo — APPROVED** [review]

File: arm-sql/2014-04-01/swagger/metrics.json (hunk 3 above)
Comment (olydis) [inline arm-sql/2014-04-01/swagger/metrics.json]: @nathannfan did you think about this?

Comment (nathannfan) [inline arm-sql/2014-04-01/swagger/metrics.json]: My bad.  I thought you meant tagging the properties in the $ref'd definitions "required," not the $ref themselves

**Comment by nathannfan** [comment]
@olydis thanks, Johannes!

**Pull request merged**